editor/src/core/es-modules/package-manager/group-component.tsx

```diff
@@ -0,0 +1,181 @@
+// TODO move me to utopia-api once stable!
+
+import * as React from 'react'
+import { CSSProperties } from 'react'
+
+const CSS_OM_SUPPORTED: boolean = document.querySelector('body')?.computedStyleMap != null
+
+interface Pins {
+  left: number
+  top: number
+  right: number
+  bottom: number
+  width: number
+  height: number
+}
+
+interface Size {
+  width: number
+  height: number
+}
+
+function isAutoOrPercentage(typedOmValue: CSSStyleValue | undefined): boolean {
+  if (typedOmValue == null) {
+    return true
+  }
+
+  if (typedOmValue instanceof CSSUnitValue && typedOmValue.unit !== 'percent') {
+    // for now the only happy case if the value is a CSSUnitValue and the unit is not percent
+    return false
+  }
+
+  // TODO recurse into compound values and check for percents
+  return true
+}
+
+function getPinPixelOrZeroIfAutoOrPercentage(
+  computedValue: string,
+  typedOmValue: CSSStyleValue | undefined,
+): number {
+  if (isAutoOrPercentage(typedOmValue)) {
+    return 0
+  }
+  if (!computedValue.endsWith('px')) {
+    return 0
+  }
+  return parseFloat(computedValue)
+}
+
+function getAllPinsSanitized(child: HTMLElement): Pins {
+  const computedStyle = window.getComputedStyle(child)
+  const computedStyleMapTyped = child.computedStyleMap()
+
+  if (computedStyle.position !== 'absolute') {
+    return { left: 0, top: 0, right: 0, bottom: 0, width: 0, height: 0 }
+  }
+
+  return {
+    left: getPinPixelOrZeroIfAutoOrPercentage(
+      computedStyle.left,
+      computedStyleMapTyped.get('left'),
+    ),
+    top: getPinPixelOrZeroIfAutoOrPercentage(computedStyle.top, computedStyleMapTyped.get('top')),
+    right: getPinPixelOrZeroIfAutoOrPercentage(
+      computedStyle.right,
+      computedStyleMapTyped.get('right'),
+    ),
+    bottom: getPinPixelOrZeroIfAutoOrPercentage(
+      computedStyle.bottom,
+      computedStyleMapTyped.get('bottom'),
+    ),
+    width: getPinPixelOrZeroIfAutoOrPercentage(
+      computedStyle.width,
+      computedStyleMapTyped.get('width'),
+    ),
+    height: getPinPixelOrZeroIfAutoOrPercentage(
+      computedStyle.width,
+      computedStyleMapTyped.get('height'),
+    ),
+  }
+}
+
+function getGroupSize(children: NodeListOf<ChildNode>): Size {
+  let maxWidth = 0
+  let maxHeight = 0
+  let maxNegativeLeft = 0
+  let maxNegativeRight = 0
+  let maxNegativeTop = 0
+  let maxNegativeBottom = 0
+
+  children.forEach((child) => {
+    if (!(child instanceof HTMLElement)) {
+      return
+    }
+
+    const sanitizedPins = getAllPinsSanitized(child)
+
+    const width = sanitizedPins.left + sanitizedPins.width + sanitizedPins.right
```

Comment:
I think this would be clearer if it was renamed to `groupWidth`, because I can't help but think it relates to this particular child.